modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -175,11 +175,30 @@
 
 (defmethod driver/describe-table :bigquery-cloud-sdk
   [_ database {table-name :name, dataset-id :schema}]
-  {:schema dataset-id
-   :name   table-name
-   :fields (-> (.. (get-table database dataset-id table-name) getDefinition getSchema)
-               table-schema->metabase-field-info
-               set)})
+  (let [table                  (get-table database dataset-id table-name)
+        tabledef               (.getDefinition table)
+        ;; a table can only have one partitioned field
+        partitioned-field-name (some-> (or (.getTimePartitioning tabledef)
+                                           (.getRangePartitioning tabledef))
+                                       .getField)
+        fields                 (set
+                                (map
+                                 #(assoc % :database-partitioned (= (:name %) partitioned-field-name))
+                                 (table-schema->metabase-field-info (. tabledef getSchema))))]
+    {:schema dataset-id
+     :name   table-name
+     :fields (cond-> fields
+               ;; if table has time partition but no field is specified as partitioned
+               ;; meaning this table is partitioned by ingestion time
+               ;; so we manually sync this virtual column
+               (and (some? (.getTimePartitioning tabledef))
+                    (nil? partitioned-field-name))
+               (conj
+                {:name                 "_PARTITIONTIME"
+                 :database-type        "TIMESTAMP"
+                 :base-type            :type/DateTimeWithLocalTZ
+                 :database-position    (count fields)
+                 :database-partitioned true}))}))
```

Comment:
Based on [the docs](https://cloud.google.com/bigquery/docs/querying-partitioned-tables#query_an_ingestion-time_partitioned_table), it looks like you also need to add a pseudo-column for `_PARTITIONDATE`.

Reply:
`_PARTITIONDATE` is `_PARTITIONTIME` truncated to DATE. so I think we can sync only `_PARTITIONTIME`, for cases where we need `_PARTITIONDATE`, we can just truncate the time column.

> If the partition granularity is daily, the table also contains a pseudo-column named _PARTITIONDATE. The value is equal to _PARTITIONTIME truncated to a DATE value.

([ref](https://cloud.google.com/bigquery/docs/querying-partitioned-tables#query_an_ingestion-time_partitioned_table))

Reply:
Makes sense, it just seemed a little strange to include one and not the other. What's the reason against doing so?

Reply:
>  What's the reason against doing so?

It's more about syncing one of the would serve our use cases. And since `_PARTITIONTIME` can be truncated to `DATE` so any use case that you need with `_PARTITIONDATE` can be handled by `_PARTITIONTIME`.

> One reason to include it is it is possible to use _PARTITIONDATE in native SQL, but it won't show in the data reference pane.

that might be a good reason to include it I think, this way, if you have the model, you can be sure you can map to the underlined column.

@luizarakaki do you think it makes sense to sync both of these pseudo columns?

Reply:
The only reason to sync `_PARTITIONDATE` is for SQL queries.

If this doesn't impact our ability to enforce filters on `_PARTITIONTIME` and doesn't have other side-effects, I think it is a good idea to have a complete data reference. If this brings problems, I wouldn't add it.

My main concern is: if the table requires filtering on partition, having both columns would bring problems? If not, I think it is useful to add `_PARTITIONDATE` too.

Reply:
The only downside I can think of is this will break the assumption that a table has only one partition field.

Reply:
One downside here is that it'll break the assumption that a table has only one partitioned field.
In milestone 3 where I'll use these filter fields for sync field values and fingerprint of a partitioned table, this assumption will come in handy.

That's said if we don't have a strong reason to add this, I'd say let's not add it for now. We can always add it later if necessary.

@calherries thoughts?

Reply:
> this assumption will come in handy.

Can you explain this more?

I had a brief look at [milestone 3](https://github.com/metabase/metabase/pull/37029) and couldn't find a reason why you can't have two fields with `:database-partitioned true`

Reply:
in this [part](https://github.com/metabase/metabase/pull/37029/files#diff-a3009f2547aa68da91de932bd775af0ce4df453500aff03c0fce30e1b6647171R42-R45) where I select the field I used as filtering field, it'd be nice to know for sure which field will be selected.

Tbh, it's not strictly needed, we can add a clause to that query to make sure either `PARTITIONTIME` or `PARTITIONDATE` is returned.

Reply:
alright I've synced them now